lib/pbio/drv/usb/usb_common_desc.c

```diff
@@ -0,0 +1,123 @@
+    .s = {
+        .desc_set_hdr = {
+            .wLength = sizeof(pbdrv_usb_ms_20_desc_set_header_t),
+            .wDescriptorType = MS_OS_20_SET_HEADER_DESCRIPTOR,
+            .dwWindowsVersion = MS_WINDOWS_VERSION_81,
+            .wTotalLength = sizeof(pbdrv_usb_ms_20_desc_set_t),
+        },
+        .compatible = {
+            .wLength = sizeof(pbdrv_usb_ms_20_compatible_t),
+            .wDescriptorType = MS_OS_20_FEATURE_COMPATBLE_ID,
+            .CompatibleID = "WINUSB",
+            .SubCompatibleID = "",
+        },
+        .reg_prop_hdr = {
+            .wLength = sizeof(pbdrv_usb_ms_20_reg_prop_hdr_t) + MS_20_REGISTRY_DATA_EXTRA_SZ,
+            .wDescriptorType = MS_OS_20_FEATURE_REG_PROPERTY,
+            .wPropertyDataType = MS_OS_20_REG_PROP_TYPE_REG_MULTI_SZ,
+        },
+        .property_name_len = sizeof(PBDRV_USB_MS_20_REG_PROPERTY_NAME),
+        .device_interface_guids = PBDRV_USB_MS_20_REG_PROPERTY_NAME,
+        .property_val_len = sizeof(PBDRV_USB_MS_20_REG_PROPERTY_VAL),
+        .device_interface_guid_val = PBDRV_USB_MS_20_REG_PROPERTY_VAL,
+    }
+};
+
+const pbdrv_usb_bos_desc_set_union_t pbdrv_usb_bos_desc_set = {
+    .s = {
+        .bos = {
+            .bLength = sizeof(pbdrv_usb_bos_desc_t),
+            .bDescriptorType = DESC_TYPE_BOS,
+            .wTotalLength = sizeof(pbdrv_usb_bos_desc_set_t),
+            .bNumDeviceCaps = 2,
+        },
+        .webusb = {
+            .hdr = {
+                .bLength = sizeof(pbdrv_usb_webusb_capability_t),
+                .bDescriptorType = DESC_TYPE_DEVICE_CAPABILITY,
+                .bDevCapabilityType = USB_DEVICE_CAPABILITY_TYPE_PLATFORM,
+                .bReserved = 0,
+                .uuid = USB_PLATFORM_CAP_GUID_WEBUSB,
+            },
+            .bcdVersion = 0x0100,
+            .bVendorCode = PBDRV_USB_VENDOR_REQ_WEBUSB,
+            .iLandingPage = PBDRV_USB_WEBUSB_LANDING_PAGE_URL_IDX,
+        },
+        .ms_20 = {
+            .hdr = {
+                .bLength = sizeof(pbdrv_usb_microsoft_20_capability_t),
+                .bDescriptorType = DESC_TYPE_DEVICE_CAPABILITY,
+                .bDevCapabilityType = USB_DEVICE_CAPABILITY_TYPE_PLATFORM,
+                .bReserved = 0,
+                .uuid = USB_PLATFORM_CAP_GUID_MS_20,
+            },
+            .dwWindowsVersion = MS_WINDOWS_VERSION_81,
+            .wMSOSDescriptorSetTotalLength = sizeof(pbdrv_usb_ms_20_desc_set_t),
+            .bMS_VendorCode = PBDRV_USB_VENDOR_REQ_MS_20,
+            .bAltEnumCode = 0,
+        }
+    }
+};
+
+const pbdrv_usb_langid_union_t pbdrv_usb_str_desc_langid = {
+    .s = {
+        .bLength = 4,
+        .bDescriptorType = DESC_TYPE_STRING,
+        .langID = {PBDRV_USB_STRING_LANGID_EN_US},
+    }
+};
+
+const pbdrv_usb_str_mfg_union_t pbdrv_usb_str_desc_mfg = {
+    .s = {
+        // Note: strip trailing null terminator
+        .bLength = sizeof(pbdrv_usb_str_mfg_t) - 2,
```

Comment:
```suggestion
        .bLength = sizeof(pbdrv_usb_str_mfg_t),
```

Let's just make the struct the right size instead by subtracting the zero-terminator there.